src/pages/signup-affiliates/components/_account-personal-details.tsx

```diff
@@ -0,0 +1,333 @@
+        ? username &&
+          first_name &&
+          last_name &&
+          date_birth &&
+          password &&
+          !first_name_error_msg &&
+          !last_name_error_msg &&
+          !password_error_msg
+        : first_name &&
+          last_name &&
+          date_birth &&
+          password &&
+          company_name &&
+          company_registration_number &&
+          !first_name_error_msg &&
+          !last_name_error_msg &&
+          !password_error_msg &&
+          !company_name_error_msg &&
+          !company_registration_error_msg
+
+    useEffect(() => {
+        updateData({
+            ...affiliate_data,
+            username,
+            first_name,
+            last_name,
+            date_birth,
+            company_name,
+            company_registration_number,
+            website_url,
+            social_media_url,
+            password,
+        })
+    }, [
+        username,
+        first_name,
+        last_name,
+        date_birth,
+        company_name,
+        company_registration_number,
+        website_url,
+        social_media_url,
+        password,
+    ])
+
+    useEffect(() => {
+        onValidate(validate)
+    }, [onValidate, validate])
+    const getFormFields = () => {
+        return form_inputs.filter((item) => {
+            if (is_individual) {
+                const company_details = ['company_name', 'company_registration_number']
+
+                return !company_details.includes(item.name)
+            }
+            return true
+        })
+    }
+    const handleInput = (e) => {
+        const { name, value } = e.target
+        switch (name) {
+            case 'username': {
+                setUsername(value)
+                return setUserNameErrorMsg(affiliate_validation.username(value))
+            }
+            case 'first_name': {
+                setFirstName(value)
+                return setFirstNameErrorMsg(affiliate_validation.first_name(value))
+            }
+            case 'last_name': {
+                setLastName(value)
+                return setLastNameErrorMsg(affiliate_validation.last_name(value))
+            }
+            case 'date_birth': {
+                setDateBirth(value)
+                break
+            }
+            case 'company_name': {
+                setCompanyName(value)
+                return setCompanyNameErrorMsg(affiliate_validation.company_name(value))
+            }
+            case 'company_registration_number': {
+                setCompanyRegistrationNumber(value)
+                return setCompanyRegistrationErrorMsg(
+                    affiliate_validation.company_registration_number(value),
+                )
+            }
+            case 'website_url': {
+                setWebsiteUrl(value)
+                return setWebsiteUrlErrorMsg(affiliate_validation.website_url(value))
+            }
+            case 'social_media_url': {
+                setSocialMedia(value)
```

Comment:
same here seems we have ts errors 

![Screenshot 2023-09-27 at 2 56 09 PM](https://github.com/binary-com/deriv-com/assets/64970259/80f4568f-7e9b-47ab-848c-ce0c9ee34630)

Reply:
resolved :)